results/cmd/watcher/main_test.go

```diff
@@ -0,0 +1,107 @@
+package main
+
+import (
+	"context"
+	"net"
+	"strings"
+	"testing"
+
+	"github.com/google/go-cmp/cmp"
+	"github.com/tektoncd/pipeline/pkg/apis/pipeline/v1beta1"
+	ttesting "github.com/tektoncd/pipeline/pkg/reconciler/testing"
+	test "github.com/tektoncd/pipeline/test"
+
+	_ "github.com/mattn/go-sqlite3"
+	"github.com/tektoncd/experimental/results/pkg/server"
+	pb "github.com/tektoncd/experimental/results/proto/proto"
+	"google.golang.org/grpc"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"knative.dev/pkg/configmap"
+)
+
+const (
+	port = ":50051"
+)
+
+func getRunName(tr *v1beta1.TaskRun) string {
+	return strings.Join([]string{tr.Namespace, tr.Name}, "/")
+}
+
+// getFakeClients create a fake client to send test data to reconciler
+func getFakeClients(t *testing.T, tr []*v1beta1.TaskRun, conn *grpc.ClientConn) (test.Assets, context.CancelFunc) {
+	t.Helper()
+	ctx, _ := ttesting.SetupFakeContext(t)
+	ctx, cancel := context.WithCancel(ctx)
+	d := test.Data{
+		TaskRuns: tr,
+	}
+	clients, _ := test.SeedTestData(t, ctx, d)
+	cmw := configmap.NewInformedWatcher(clients.Kube, "")
+	return test.Assets{
+		Controller: newController(ctx, cmw, conn),
+		Clients:    clients,
+	}, cancel
+}
+
+// reconcileTaskRun sends TaskRun data to reconciler and then retrieves completed TaskRun
+func reconcileTaskRun(taskRun v1beta1.TaskRun, d []*v1beta1.TaskRun, t *testing.T, conn *grpc.ClientConn) (*v1beta1.TaskRun, error) {
+	testAssets, cancel := getFakeClients(t, d, conn)
+	defer cancel()
+	c := testAssets.Controller
+	clients := testAssets.Clients
+	ctx := context.Background()
+	if err := c.Reconciler.Reconcile(ctx, getRunName(&taskRun)); err != nil {
+		return nil, err
+	}
+	tr, err := clients.Pipeline.TektonV1beta1().TaskRuns(taskRun.Namespace).Get(taskRun.Name, metav1.GetOptions{})
+	if err != nil {
+		return nil, err
+	}
+	return tr, nil
+}
+
+// TestReconcile tests if TaskRun in the client can be properly updated when sent to reconciler
+func TestReconcile(t *testing.T) {
+	srv, err := server.SetupTestDB(t)
+	if err != nil {
+		t.Fatalf("Failed to create fake server: %v", err)
+	}
+	s := grpc.NewServer()
+	pb.RegisterResultsServer(s, srv) // local test server
+	lis, err := net.Listen("tcp", port)
+	if err != nil {
+		//
+	}
+	defer lis.Close()
```

Comment:
minor: might be useful to extract the lines that setup the grpc server to its own helper function.

Reply:
done